libs/community/langchain_community/chat_models/notdiamond.py

```diff
@@ -0,0 +1,416 @@
+        }
+        return self._create_chat_result(result)
+
+    def _stream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[ChatGenerationChunk]:
+        message_dicts, params = self._create_message_dicts(messages, stop)
+        params = {**params, **kwargs, "stream": True}
+
+        generator_response = completion_with_retry(
+            self,
+            messages=message_dicts,
+            run_manager=run_manager,
+            **params,
+        )
+        for chunk in generator_response:
+            if not chunk.content:
+                continue
+            cg_chunk = ChatGenerationChunk(message=chunk)
+            if run_manager:
+                run_manager.on_llm_new_token(chunk.content, chunk=cg_chunk)
+            yield cg_chunk
+
+    async def _astream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[ChatGenerationChunk]:
+        message_dicts, params = self._create_message_dicts(messages, stop)
+        params = {**params, **kwargs, "stream": True}
+
+        generator_response = completion_with_retry(
+            self,
+            messages=message_dicts,
+            run_manager=run_manager,
+            **params,
+        )
+        async for chunk in await generator_response:
+            if not chunk.content:
+                continue
+            cg_chunk = ChatGenerationChunk(message=chunk)
+            if run_manager:
+                await run_manager.on_llm_new_token(chunk.content, chunk=cg_chunk)
+            yield cg_chunk
+
+    def _create_chat_result(self, response: Mapping[str, Any]) -> ChatResult:
+        res = response['response']
+        gen = ChatGeneration(
+            message=AIMessage(content=res.content),
+            generation_info=dict(finish_reason=res.response_metadata.get("finish_reason")),
+        )
+        input_tokens = res.usage_metadata.get("input_tokens", {})
+        output_tokens = res.usage_metadata.get("output_tokens", {})
+        total_tokens = res.usage_metadata.get("total_tokens", {})
+        session_id = response['session_id']
+        provider = response['provider']
+        llm_output = {
+            "session_id": session_id,
+            "model_name": f"{provider.provider}/{provider.model}",
+            "input_tokens": input_tokens,
+            "output_tokens": output_tokens,
+            "total_tokens": total_tokens,
+        }
+        return ChatResult(generations=[gen], llm_output=llm_output)
+
+    def _create_message_dicts(
+        self, messages: List[BaseMessage], stop: Optional[List[str]]
+    ) -> Tuple[List[Dict[str, Any]], Dict[str, Any]]:
+        params = self._default_params
+        if stop is not None:
```

Comment:
Let's catch the possibility that `stop` is `False`-ish since it can be an empty list

```suggestion
        if not stop:
```